.travis.yml

```diff
@@ -3,23 +3,71 @@ node_js:
   - lts/*
 git:
   depth: 3
+addons:
+  apt:
+    packages:
+      # Needed hack to have colored output, see: travis-ci/travis-ci/#7967
+      # This intstalls unbuffer which reenables colored output for builds
+      # with secret environment variables
+      - expect-dev
 cache:
   - pip
   - npm
-install:
-  - pip install grow==0.7.4 --user
-  - grow --version
-  - which grow
-  - npm ci
-before_deploy:
-  - npm run build:staging
-deploy:
-  provider: gae
-  keyfile: ".staging.key.json"
-  project: amp-dev-staging
-  skip_cleanup: true
-  on:
-    branch: future
 before_install:
 - openssl aes-256-cbc -K $encrypted_d4a09416a845_key -iv $encrypted_d4a09416a845_iv
   -in .staging.key.json.enc -out .staging.key.json -d
+- gcloud auth activate-service-account --key-file=.staging.key.json
+install:
+  - npm ci
+stages:
+  - setup
+  - build
+  - deploy
+jobs:
+  include:
+    - stage: setup
+      name: 'Check code, import documents, build samples, playground & boilerplate'
+      script: unbuffer gulp setupBuild
+    - &build
+      stage: build
+      install:
+        - npm ci
+        - pip install grow==0.7.4 --user
+      name: 'Build Pages (EN)'
+      script: unbuffer gulp buildPages --locales en
+    - <<: *build
+      name: 'Build Pages (FR)'
+      script: unbuffer gulp buildPages --locales fr
+    - <<: *build
+      name: 'Build Pages (AR)'
+      script: unbuffer gulp buildPages --locales ar
+    - <<: *build
+      name: 'Build Pages (ES)'
+      script: unbuffer gulp buildPages --locales es
+    - <<: *build
+      name: 'Build Pages (IT)'
+      script: unbuffer gulp buildPages --locales it
+    - <<: *build
+      name: 'Build Pages (ID)'
+      script: unbuffer gulp buildPages --locales id
+    - <<: *build
+      name: 'Build Pages (JA)'
+      script: unbuffer gulp buildPages --locales ja
+    - <<: *build
+      name: 'Build Pages (ko)'
+      script: unbuffer gulp buildPages --locales ko
+    - <<: *build
+      name: 'Build Pages (BR)'
+      script: unbuffer gulp buildPages --locales pt_BR
+    - <<: *build
+      name: 'Build Pages (RU)'
+      script: unbuffer gulp buildPages --locales ru
+    - <<: *build
+      name: 'Build Pages (TR)'
+      script: unbuffer gulp buildPages --locales tr
+    - <<: *build
+      name: 'Build Pages (CN)'
+      script: unbuffer gulp buildPages --locales zh_CN
```

Comment:
Is it possible to auto generate travis jobs? I don't like hard-coding the languages here.

Reply:
I agree. There's no way to seed jobs from a different source than .travis.yml so I thought about using a [build matrix](https://docs.travis-ci.com/user/build-matrix/) but that [didn't work out](https://travis-ci.org/ampproject/docs/builds/532377988). Somehow the expansion always gets assigned to the Test stage.

Reply:
OK, let's keep it like this for now then.